packages/react/src/components/RadioButtonGroup/RadioButtonGroup.js

```diff
@@ -99,16 +98,13 @@ export default class RadioButtonGroup extends React.Component {
         );
       }
 
-      return (
-        <RadioButton
-          {...other}
-          name={this.props.name}
-          key={value}
-          value={value}
-          onChange={this.handleChange}
-          checked={value === this.state.selected}
-        />
-      );
+      return React.cloneElement(radioButton, {
+        name: this.props.name,
+        key: value,
+        value: value,
+        onChange: this.handleChange,
+        checked: value === this.state.selected,
+      });
```

Comment:
Since all the children may not be `RadioButton` elements, should we check if they are before adding irrelevant data to these elements?

```jsx
      if (radioButton.type.displayName === 'RadioButton') {
        return React.cloneElement(radioButton, {
          name: this.props.name,
          key: value,
          value: value,
          onChange: this.handleChange,
          checked: value === this.state.selected,
        });
      } else {
        return radioButton;
      }
```

Otherwise, I'm seeing the `name` attribute being placed on a custom `div` I rendered adjacent to the first RadioButton

<img width="393" alt="Screen Shot 2021-04-05 at 4 33 24 PM" src="https://user-images.githubusercontent.com/11928039/113623776-cd2df280-962c-11eb-8242-dc18d71bfef2.png">

Reply:
These are props we want to override.

Thoughts @vpicone 👀

Reply:
In my example, I had

```jsx
<RadioButtonGroup
    legendText="Radio button heading"
    name="radio-button-group"
    defaultSelected="radio-1">
      <RadioButton labelText="Option 1" value="radio-1" id="radio-1" />
      <div>A custom sublabel</div>
      <RadioButton labelText="Option 2" value="radio-2" id="radio-2" />
      <RadioButton labelText="Option 3" value="radio-3" id="radio-3" />
</RadioButtonGroup>
```

and the `div` was getting the `name` attribute. I think we would only want to apply these props to `RadioButton` elements? 

I could be misunderstanding the original issue! Do we have a different sample code block to test?

Reply:
@tw15egan the aim here is to let folks customize/wrap `RadioButton` with their own custom component. Doing a check on the `displayName` prevents that in the same way that replacing their component with our own does (current state). I don't think we can't solve the arbitrary `div` wrapper scenario without context, but we can pass their custom component props as if it were a `RadioButton` so that they can be applied in their component.

```jsx
const MySpecialRadioButton = (props) => (
  <div className="my-custom-class">
    <RadioButton {...props} />
    <p>My special description</p>
  </div>
);

const App = () => (
  <RadioButtonGroup name="radio-button-group">
    <MySpecialRadioButton
      id="radio-1"
      value="standard"
    />
    <RadioButton id="radio-2" value="standard" />
  </RadioButtonGroup>
);
```

Reply:
Makes sense! Thanks for clarifying it for me 😄